blenderproc/python/types/ArmatureUtility.py

```diff
@@ -0,0 +1,202 @@
+        super().__init__(bpy_object=bpy_object)
+
+    def set_rotation_euler(self, rotation_euler: Union[float, list, Euler, np.ndarray], mode: str = "absolute", frame: int = None):
+        """ Rotates the armature based on euler angles. Validates values with given constraints.
+
+        :param rotation_euler: The amount of rotation (in radians). Either three floats for x, y and z axes, or a single float.
+                               In the latter case, the axis of rotation is derived based on the rotation constraint. If
+                               these are not properly set (i.e., two axes must have equal min/max values) an exception will be thrown.
+        :param mode: One of ["absolute", "relative"]. For absolute rotations we clip the rotation value based on the constraints.
+                     For relative we don't - this will result in inverse motion after the constraint's limits have been reached.
+        :param frame: Keyframe where to insert the respective rotations.
+        """
+        assert mode in ["absolute", "relative"]
+        bpy.ops.object.select_all(action='DESELECT')
+        bone = self.blender_obj.pose.bones.get('Bone')
+        bone.bone.select = True
+        bone.rotation_mode = 'XYZ'
+
+        # in absolute mode we overwrite the rotation values of the armature
+        if mode == "absolute":
+            if isinstance(rotation_euler, float):
+                axis = self._determine_rotation_axis()
+                rotation_euler = self._clip_value_from_constraint(value=rotation_euler, constraint_name="Limit Rotation", axis=axis)
+                current_rotation_euler = bone.rotation_euler
+                current_rotation_euler[["X", "Y", "Z"].index(axis)] = rotation_euler
+                bone.rotation_euler = current_rotation_euler
+                print(f"Set rotation_euler of armature {self.get_name()} to {rotation_euler}")
+            else:
+                bone.rotation_euler = Vector([self._clip_value_from_constraint(value=rot_euler, constraint_name="Limit Rotation", axis=axis)
+                                       for rot_euler, axis in zip(rotation_euler, ["X", "Y", "Z"])])
+                print(f"Set rotation_euler of armature {self.get_name()} to {rotation_euler}")
+        # in relative mode we add the rotation to the current value
+        elif mode == "relative":
+            if isinstance(rotation_euler, float):
+                axis = self._determine_rotation_axis()
+                bone.rotation_euler.rotate_axis(axis, rotation_euler)
+                print(f"Relatively rotated armature {self.get_name()} around axis {axis} for {rotation_euler} radians")
+            else:
+                for axis, rotation in zip(["X", "Y", "Z"], rotation_euler):
+                    bone.rotation_euler.rotate_axis(axis, rotation)
+                print(f"Relatively rotated armature {self.get_name()} for {rotation_euler} radians")
+
+        Utility.insert_keyframe(bone, "rotation_euler", frame)
```

Comment:
Can we maybe put in here the frame_end check?

```suggestion
        Utility.insert_keyframe(bone, "rotation_euler", frame)
        if frame > bpy.context.scene.frame_end:
            bpy.context.scene.frame_end += 1
```